### ISSUE TITLE
Remove separate action for "Test nginx and haproxy"

### ISSUE THREAD

**Issue by garethbowen** [medic/cht-core#8708]
**Describe the issue**

We have a "Test nginx and haproxy" action which runs unit tests on our nginx and haproxy code, but it's not required so doesn't block merging, and is not part of the main "Build and test" action so it doesn't block the rest of the testing from running.

**Describe the improvement you'd like**

Make the unit tests run with the rest of the unit tests in the "Compile the app" job in the "Build and test" action. This will make them required and skip the rest of the build if they fail.

**Describe alternatives you've considered**

- We could just mark the "Test nginx and haproxy" action as required, but that doesn't stop the e2e tests from running which is very expensive.
- One downside of this approach is it'll slow down the "Build and test" action by maybe 1 minute. Another option is to make the action a new step that runs in parallel with "Compile the app" which is a little wasteful in terms of firing up two GH runners and checking out the code twice, but would be faster. If we go down this route there are other parts that could be split out from "Compile the app" which would save even more time.

**Comment by garethbowen** [comment]
@nydr I'd appreciate your input here!

**Comment by nydr** [comment]
Looking at [past failures](https://github.com/medic/cht-core/actions/workflows/side-service-test.yml?query=is%3Afailure) I see mainly failures due to docker repo or auth outages and ideally those would not be blocking a merge. Based on this I don't think these tests are suitable as inexpensive smoke tests.

When it comes to making these tests blocking or not I don't have strong opinions. I feel that the people who would be modifying these are responsible enough to be able to make a judgement call if it can be merged despite failures. I'm slightly concerned with the outage related failures that are unrelated to changes as these files are rarely changed, but since it's not very common and they are quick to rerun I think blocking is fine as well.

#### For cost savings I see two possible actions
1. Run tests concurrently and have one or more fast/inexpensive smoke test tasks that other tests [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds)
   - Ideally there would be some that's fast enough to run locally as a git pre-commit [hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
2. Configure haproxy and nginx tests to only be run on PRs where relevant files are changed using [paths](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths)

**Comment by nydr** [comment]
> All this makes me think we should just run it at the end of the current build and unit test build. Would you mind if I give this a try and see if it works? I promise not to merge without you reviewing first!

Absolutely, I'm just excited to see that someone wants to improve these builds!

> > Ideally there would be some that's fast enough to run locally as a git pre-commit [hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
> 
> The building step probably makes that too slow, right?

Agreed, I was thinking running a subset of the tests locally before building the image. I aim to keep pre-commit hooks under a second, ideally just a frame or so, but that might be unrealistic in most node projects. Locally I configured eslint to run on commits but it's just around 1s so I'm not sure that's a good idea.

**Comment by garethbowen** [comment]
Merged.
